src/main/java/com/hms_networks/americas/sc/tagupdateservice/TagUpdateHandler.java

```diff
@@ -0,0 +1,578 @@
+  private static final String TAG_UPDATE_MESSAGE_TAG_NAME_KEY = "name";
+
+  /** Constant string value used to access the tag type field in tag update messages. */
+  private static final String TAG_UPDATE_MESSAGE_TAG_TYPE_KEY = "type";
+
+  /** Constant string value used to access the tag value field in tag update messages. */
+  private static final String TAG_UPDATE_MESSAGE_TAG_VALUE_KEY = "value";
+
+  /**
+   * Default boolean value of the restore previous values on fault field in tag update messages, if
+   * not present.
+   */
+  private static final boolean TAG_UPDATE_MESSAGE_RESTORE_PREVIOUS_ON_FAULT_DEFAULT = false;
+
+  /** Constant string value used to indicate the integer tag type. */
+  private static final String TAG_UPDATE_MESSAGE_TAG_TYPE_INTEGER_STRING = "integer";
+
+  /** Constant string value used to indicate the float tag type. */
+  private static final String TAG_UPDATE_MESSAGE_TAG_TYPE_FLOAT_STRING = "float";
+
+  /** Constant string value used to indicate the string tag type. */
+  private static final String TAG_UPDATE_MESSAGE_TAG_TYPE_STRING_STRING = "string";
+
+  /** Constant string value used to indicate the boolean tag type. */
+  private static final String TAG_UPDATE_MESSAGE_TAG_TYPE_BOOLEAN_STRING = "boolean";
+
+  /** Constant string value used to indicate the DWORD tag type. */
+  private static final String TAG_UPDATE_MESSAGE_TAG_TYPE_DWORD_STRING = "dword";
+
+  /**
+   * Applies the tag values from the specified {@link JSONArray} of tag values, and restores the
+   * original tag values if a failure occurs as specified by the parameter <code>
+   * restorePreviousTagValuesOnUpdateFailure</code>.
+   *
+   * @param tagValuesArray array of tag values to update/apply
+   * @param restorePreviousTagValuesOnUpdateFailure boolean indicating if original tag values should
+   *     be restored on error
+   * @return integer enumeration indicating result (apply failure, type mismatch, etc)
+   */
+  private static TagUpdateResult applyTagValuesFromJsonArray(
+      JSONArray tagValuesArray, boolean restorePreviousTagValuesOnUpdateFailure) {
+    // Create map to store previous values in case of restore, if enabled
+    HashMap previousValsMap = null;
+    if (restorePreviousTagValuesOnUpdateFailure) {
+      previousValsMap = new HashMap();
+    }
+
+    // Loop through each tag in array
+    TagUpdateResult tagUpdateResult = TagUpdateResult.SUCCESS;
+    String currTagName = "";
+    String currTagValueString = "";
+    for (int tagIndex = 0; tagIndex < tagValuesArray.length(); tagIndex++) {
+      try {
+        // Get tag object
+        JSONObject tagObject = tagValuesArray.getJSONObject(tagIndex);
+        String currTagType = tagObject.getString(TAG_UPDATE_MESSAGE_TAG_TYPE_KEY);
+        currTagName = tagObject.getString(TAG_UPDATE_MESSAGE_TAG_NAME_KEY);
+
+        // Create tag control object
+        TagControl currTagControl = new TagControl();
```

Comment:
Would it be slightly quicker to initialize curTagControl object outside scope of for loop?  If it's inside, is it created and garbage collected for each iteration of the loop?

Reply:
Per our offline conversation, this will remain as is, but further investigation (in general) is warranted to evaluate this potential performance and functionality difference.